.github/workflows/semver-checks-pr-label.yml

```diff
@@ -0,0 +1,87 @@
+on:   # yamllint disable-line rule:truthy
+  workflow_run:
+    workflows: [Check semver breaks]
+    types: [completed]
+
+name: Check semver breaks - Label and Comment PR
+
+jobs:
+  Download:
+    name: Download, Unzip and Add Labels/Comments
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+    # only run if CI passes on the "Check semver breaks" workflow
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    steps:
+      - name: "Download artifact"
+        uses: actions/github-script@v7
+        with:
+          script: |
+            // get all artifacts from the workflow run
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+
+            // find the artifact that starts with 'semver-break'
+            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name.startsWith('semver-break');
+            })[0];
+
+            // if no artifact found, exit
+            if (!matchArtifact) {
+              console.log('No semver-break artifact found');
+              process.exit(0);
+            }
+
+            // otherwise download the artifact
+            let download = await github.rest.actions.downloadArtifact({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              artifact_id: matchArtifact.id,
+              archive_format: 'zip',
+            });
+
+            // write the artifact to the workspace
+            let fs = require('fs');
+            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/semver-break.zip`, Buffer.from(download.data));
+      - name: "Unzip artifact"
+        if: ${{ hashFiles('semver-break.zip') != '' }}
+        run: unzip semver-break.zip
+      - name: "Save PR number to env"
+        if: ${{ hashFiles('semver-break') != '' }}
+        run: |
+          PR_NUMBER=$(cat semver-break)
+          # check if PR_NUMBER is a number
+          if ! [[ $PR_NUMBER =~ ^-?[0-9]+$ ]]; then
+              echo "$PR_NUMBER is not a number."
+              exit 1
+          fi
+          echo "PR_NUMBER=$PR_NUMBER" >> $GITHUB_ENV
+      - name: "Comment and add label on PR - Semver break"
+        uses: actions/github-script@v7
+        if: ${{ hashFiles('semver-break') != '' }}
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            // get the PR number from the env
+            var issue_number = Number(${{ env.PR_NUMBER }});
```

Comment:
This is pretty risky. If not for the earlier sanitization, this is _definitely_ injectable.

Consider reading it from a file instead of templating it in like this. This feels like a footgun ready to go off at any second.

Reply:
Oh yeah, reading in JS directly and sanitizing in it might be safer. Though if the sanitization is kept it's not a big deal.

Reply:
Now it is sanitized in JS land as well:
https://github.com/rust-bitcoin/rust-bitcoin/blob/2ed8f46bce3653e8d93cada9780dff3349382279/.github/workflows/semver-checks-pr-label.yml#L69-L78

Reply:
We mean you could just load the file directly.

Reply:
Ok, now its

https://github.com/rust-bitcoin/rust-bitcoin/blob/6ab8f99731fb8b4741222c5156ffe6c2ebb3cc57/.github/workflows/semver-checks-pr-label.yml#L60-L69